tests/basic_integration/test.sh

```diff
@@ -0,0 +1,155 @@
+  echo "This script must be run from the root"
+  exit 1
+fi
+
+# Environment variables
+export RPC_URL="ws://localhost:8545"
+# These contracts are based on the deterministic order of hardhat deploy
+export ENCLAVE_CONTRACT="0x9fE46736679d2D9a65F0992F2272dE9f3c7fa6e0"
+export REGISTRY_CONTRACT="0xCf7Ed3AccA5a467e9e704C703E8D87F634fB0Fc9"
+# These are random addresses for now
+export CIPHERNODE_ADDRESS_1="0x2546BcD3c84621e976D8185a91A922aE77ECEc30"
+export CIPHERNODE_ADDRESS_2="0xbDA5747bFD65F08deb54cb465eB87D40e51B197E"
+export CIPHERNODE_ADDRESS_3="0xdD2FD4581271e230360230F9337D5c0430Bf44C0"
+export CIPHERNODE_ADDRESS_4="0x8626f6940E2eb28930eFb4CeF49B2d1F2C9C1199"
+
+# Function to clean up background processes
+cleanup() {
+    echo "Cleaning up processes..." >&2
+    kill $(jobs -p) 2>/dev/null
+    sleep 4
+    exit ${1:-1}
+}
+
+heading() {
+    echo ""
+    echo ""
+    echo "--------------------------------------------------------------"
+    echo " $1     "
+    echo "--------------------------------------------------------------"
+    echo ""
+}
+
+waiton() {
+    local file_path="$1"
+    until [ -f "$file_path" ]; do
+        sleep 1
+    done
+}
+
+waiton-files() {
+    while true; do
+        all_exist=true
+        for file in "$@"; do
+            if [ ! -f "$file" ]; then
+                all_exist=false
+                break
+            fi
+        done
+        if $all_exist; then
+            break
+        fi
+        sleep 1
+    done
+}
+
+pkill -9 -f "target/debug/node" || true
+pkill -9 -f "hardhat node" || true
+pkill -9 -f "target/debug/aggregator" || true
+
+# Set up trap to catch errors and interrupts
+trap 'cleanup $?' ERR INT TERM
+
+# Delete output artifacts
+rm -rf $ROOT_DIR/tests/basic_integration/output/*
+
+$SCRIPT_DIR/lib/prebuild.sh
+
+heading "Start the EVM node"
+
+yarn evm:node &
+
+until curl -f -s "http://localhost:8545" > /dev/null; do
+  sleep 1
+done
+
+heading "Launch ciphernode $CIPHERNODE_ADDRESS_1"
+yarn ciphernode:launch --address $CIPHERNODE_ADDRESS_1 --rpc "$RPC_URL" --enclave-contract $ENCLAVE_CONTRACT --registry-contract $REGISTRY_CONTRACT &
+
+heading "Launch ciphernode $CIPHERNODE_ADDRESS_2"
+yarn ciphernode:launch --address $CIPHERNODE_ADDRESS_2 --rpc "$RPC_URL" --enclave-contract $ENCLAVE_CONTRACT --registry-contract $REGISTRY_CONTRACT &
+
+heading "Launch ciphernode $CIPHERNODE_ADDRESS_3"
+yarn ciphernode:launch --address $CIPHERNODE_ADDRESS_3 --rpc "$RPC_URL" --enclave-contract $ENCLAVE_CONTRACT --registry-contract $REGISTRY_CONTRACT &
+
+heading "Launch ciphernode $CIPHERNODE_ADDRESS_4"
+yarn ciphernode:launch --address $CIPHERNODE_ADDRESS_4 --rpc "$RPC_URL" --enclave-contract $ENCLAVE_CONTRACT --registry-contract $REGISTRY_CONTRACT &
+
+yarn ciphernode:aggregator --rpc "$RPC_URL" --enclave-contract $ENCLAVE_CONTRACT --registry-contract $REGISTRY_CONTRACT --pubkey-write-path "$SCRIPT_DIR/output/pubkey.bin" --plaintext-write-path "$SCRIPT_DIR/output/plaintext.txt" &
+
+sleep 1
+
```

Comment:
We need to wait for the binaries to launch and don;t want to write a healthcheck right now - wont fix